src/equations/compressible_euler_quasi_1d.jl

```diff
@@ -0,0 +1,368 @@
+    # Unpack left and right state
+    rho_ll, v1_ll, p_ll, a_ll = cons2prim(u_ll, equations)
+    rho_rr, v1_rr, p_rr, a_rr = cons2prim(u_rr, equations)
+
+    # Compute the necessary mean values
+    rho_mean = Trixi.ln_mean(rho_ll, rho_rr)
+    # Algebraically equivalent to `inv_ln_mean(rho_ll / p_ll, rho_rr / p_rr)`
+    # in exact arithmetic since
+    #     log((ϱₗ/pₗ) / (ϱᵣ/pᵣ)) / (ϱₗ/pₗ - ϱᵣ/pᵣ)
+    #   = pₗ pᵣ log((ϱₗ pᵣ) / (ϱᵣ pₗ)) / (ϱₗ pᵣ - ϱᵣ pₗ)
+    inv_rho_p_mean = p_ll * p_rr * Trixi.inv_ln_mean(rho_ll * p_rr, rho_rr * p_ll)
+    v1_avg = 0.5 * (v1_ll + v1_rr)
+    a_v1_avg = 0.5 * (a_ll * v1_ll + a_rr * v1_rr)
+    p_avg = 0.5 * (p_ll + p_rr)
+    velocity_square_avg = 0.5 * (v1_ll * v1_rr)
+
+    # Calculate fluxes
+    # Ignore orientation since it is always "1" in 1D
+    f1 = rho_mean * a_v1_avg
+    f2 = rho_mean * a_v1_avg * v1_avg
+    f3 = f1 * (velocity_square_avg + inv_rho_p_mean * equations.inv_gamma_minus_one) +
+         0.5 * (p_ll * a_rr * v1_rr + p_rr * a_ll * v1_ll)
+
+    return SVector(f1, f2, f3, zero(eltype(u_ll)))
+end
+
+# Calculate estimates for maximum wave speed for local Lax-Friedrichs-type dissipation as the
+# maximum velocity magnitude plus the maximum speed of sound
+@inline function max_abs_speed_naive(u_ll, u_rr, orientation::Integer,
+                                     equations::CompressibleEulerEquationsQuasi1D)
+    a_rho_ll, a_rho_v1_ll, a_e_ll, a_ll = u_ll
+    a_rho_rr, a_rho_v1_rr, a_e_rr, a_rr = u_rr
+
+    # Calculate primitive variables and speed of sound
+    rho_ll = a_rho_ll / a_ll
+    e_ll = a_e_ll / a_ll
+    v1_ll = a_rho_v1_ll / a_rho_ll
+    v_mag_ll = abs(v1_ll)
+    p_ll = (equations.gamma - 1) * (e_ll - 0.5 * rho_ll * v_mag_ll^2)
+    c_ll = sqrt(equations.gamma * p_ll / rho_ll)
+    rho_rr = a_rho_rr / a_rr
+    e_rr = a_e_rr / a_rr
+    v1_rr = a_rho_v1_rr / a_rho_rr
+    v_mag_rr = abs(v1_rr)
+    p_rr = (equations.gamma - 1) * (e_rr - 0.5 * rho_rr * v_mag_rr^2)
+    c_rr = sqrt(equations.gamma * p_rr / rho_rr)
+
+    λ_max = max(v_mag_ll, v_mag_rr) + max(c_ll, c_rr)
+end
+
+@inline function Trixi.max_abs_speeds(u, equations::CompressibleEulerEquationsQuasi1D)
+    a_rho, a_rho_v1, a_e, a = u
+    rho = a_rho / a
+    v1 = a_rho_v1 / a_rho
+    e = a_e / a
+    p = (equations.gamma - 1) * (e - 0.5 * rho * v1^2)
+    c = sqrt(equations.gamma * p / rho)
+
+    return (abs(v1) + c,)
+end
+
+# Convert conservative variables to primitive
+@inline function cons2prim(u, equations::CompressibleEulerEquationsQuasi1D)
+    a_rho, a_rho_v1, a_e, a = u
+    q = cons2prim(u, CompressibleEulerEquations1D(equations.gamma))
+
+    return SVector(q[1] / a, q[2], q[3] / a, a)
+end
+
+# Convert conservative variables to entropy
+@inline function Trixi.cons2entropy(u, equations::CompressibleEulerEquationsQuasi1D)
+    a_rho, a_rho_v1, a_e, a = u
+    q = cons2entropy(u, CompressibleEulerEquations1D(equations.gamma))
+
+    w1 = q[1] - log(a)
```

Comment:
Where did you get the `- log(a)` expression from? I thought the original paper stated that the entropy variables for the quasi-1D system were the same as the entropy variables of the original compressible Euler equations (for the choice of entropy we looked at).